tests/jest/unit/modules/atomic-widgets/commands-internal/delete-variant.test.js

```diff
@@ -0,0 +1,154 @@
+import createContainer from '../createContainer';
+
+describe( 'DeleteVariant - apply', () => {
+	let DeleteVariantCommand;
+
+	beforeEach( async () => {
+		global.$e = {
+			internal: jest.fn(),
+			run: jest.fn(),
+			modules: {
+				editor: {
+					CommandContainerInternalBase: class {},
+				},
+			},
+		};
+
+		// Need to import dynamically since the command extends a global variable which isn't available in regular import.
+		DeleteVariantCommand = ( await import( 'elementor/modules/atomic-widgets/assets/js/editor/commands-internal/delete-variant' ) ).default;
+	} );
+
+	afterEach( () => {
+		delete global.$e;
+		delete global.elementor;
+
+		jest.resetAllMocks();
+	} );
+
+	it( 'should throw an error if style def not exits', () => {
+		const command = new DeleteVariantCommand();
+
+		const bind = 'classes';
+		const container = createContainer( {
+			widgetType: 'a-heading',
+			elType: 'widget',
+			id: '123',
+			settings: {
+				text: 'Test text',
+				[ bind ]: {
+					$$type: 'classes',
+					value: [ 'style-id' ],
+				},
+			},
+			styles: {
+				'style-id': {
+					id: 'style-id',
+					label: '',
+					type: 'class',
+					variants: [],
+				},
+			},
+		} );
+
+		// Act & Assert
+		expect( () => {
+			command.apply( { container, styleDefId: 'not-exits-style-id', meta: { breakpoint: null, state: null } } );
+		} ).toThrowError( 'Style Def not found' );
+	} );
+
+	it( 'should delete style variant', () => {
+		const command = new DeleteVariantCommand();
+
+		const bind = 'classes';
+		const container = createContainer( {
+			widgetType: 'a-heading',
+			elType: 'widget',
+			id: '123',
+			settings: {
+				text: 'Test text',
+				[ bind ]: {
+					$$type: 'classes',
+					value: [ 'style-id' ],
+				},
+			},
+			styles: {
+				'style-id': {
+					id: 'style-id',
+					label: '',
+					type: 'class',
+					variants: [],
+				},
+			},
+		} );
+
+		// Act
+		command.apply( { container, styleDefId: 'style-id', meta: { breakpoint: null, state: null } } );
+
+		const updatedStyles = {
+			'style-id': {
+				id: 'style-id',
+				label: '',
+				type: 'class',
+				variants: [],
+			},
+		};
+
+		// Assert
+		expect( container.model.get( 'styles' ) ).toEqual( updatedStyles );
+	} );
+
+	it( 'should delete style variant without deleting other variants', () => {
+		const command = new DeleteVariantCommand();
+
+		const bind = 'classes';
+		const container = createContainer( {
+			widgetType: 'a-heading',
+			elType: 'widget',
+			id: '123',
+			settings: {
+				text: 'Test text',
+				[ bind ]: {
+					$$type: 'classes',
+					value: [ 'style-id' ],
+				},
+			},
+			styles: {
+				'style-id': {
+					id: 'style-id',
+					label: '',
+					type: 'class',
+					variants: [
+						{
+							meta: { breakpoint: null, state: null },
+							props: {},
+						},
+						{
+							meta: { breakpoint: 'sm', state: null },
+							props: {},
+						},
```

Comment:
add a case that'll fail with the current logic you have (`||` vs `&&`), and then fix the code and see that the test passes